release_docs/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,204 @@
+        - `$ git clone ​https://github.com/HDFGroup/hdf5.git`
+    - Create a release preparation branch (hdf5_X_Y_Z) by branching off of the support branch (hdf5_X_Y):
+        - `$ git checkout hdf5_X_Y`
+        - `$ git checkout -b hdf5_X_Y_Z`
+        - or create the new branch in GitHub GUI.
+    - Check that required CMake files to point to the specific versions of the third-party software (szip and zlib) that they depend on.
+        - Update as needed.
+    - Change the **support** branch to X.Y.{Z+1}-1 using the [bin/h5vers](https://github.com/HDFGroup/hdf5/blob/develop/bin/h5vers) script: 
+        - `$ git checkout hdf5_X_Y`
+        - `$ bin/h5vers -s X.Y.{Z+1}-1;`
+        - `$ git commit -m "Updated support branch version number to X.Y.{Z+1}-1"`
+        - `$ git push`
+    - Change the **release preparation branch**'s version number to X.Y.Z-{SR+1} using the [bin/h5vers](https://github.com/HDFGroup/hdf5/blob/develop/bin/h5vers) script: 
+        - `$ git checkout hdf5_X_Y_Z;` 
+        - `$ bin/h5vers -s X.Y.Z-{SR+1};` 
+        - `$ git commit -m "Updated release preparation branch version number to X.Y.Z-{SR+1}"` 
+        - `$ git push` 
+    - Update default configuration mode
+        - `$ git checkout hdf5_X_Y_Z;` and `$ bin/switch_maint_mode -disable ./configure.ac` to disable `AM_MAINTAINER_MODE`. 
+        - Need to set option `HDF5_GENERATE_HEADERS` to `OFF`, currently in line 996 of [src/CMakeLists.txt](https://github.com/HDFGroup/hdf5/blob/develop/src/CMakeLists.txt).
+        - Change the **release preparation branch**'s (i.e. hdf5_X_Y_Z) default configuration mode from development to production in [configure.ac](https://github.com/HDFGroup/hdf5/blob/develop/configure.ac). 
+        - Find “Determine build mode” in [configure.ac](https://github.com/HDFGroup/hdf5/blob/develop/configure.ac). 
+        - Change `default=debug` to `default=production` at the bottom of the `AS_HELP_STRING` for `--enable-build-mode`.
+        - Under `if test "X-$BUILD_MODE" = X- ; then` change `BUILD_MODE=debug` to `BUILD_MODE=production`. 
+        - Run `“sh ./autogen.sh”` to regenerate the UNIX build system files and commit the changes. (use `git status --ignored` to see the changes and `git add -f` to add all files. First delete any new files not to be committed, notably `src/H5public.h~` and `autom4te.cache/`.)
+        - `$ git push with commit message listing change steps for creating release branch`
+    - E-mail hdf5lib@hdfgroup.org to indicate that the code freeze on the release support branch (i.e. hdf5_X_Y) has been lifted and development on the next maintenance release can resume. The code freeze will remain in place on the release preparation branch (i.e. hdf5_X_Y_Z) indefinitely. 
+
+7. Perform Release Testing (Test Automation Team | Release Manager | Project Leads)
+    - Verify that source and binary distributions of HDF5 are acceptable on all target operating environments.
```

Comment:
Should the sub-tasks be an ordered list?

Reply:
I have updated this.